### PR TITLE
Adding warning in purge

### LINKED ISSUE #961: Not clearing DB on purge
## Acceptance Criteria

- [ ] Running `covalent purge` should warn users about deleting database, and user should confirm prior to continuing

### PR DESCRIPTION
**Pull request by kessler-frost**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

Resolves #961 
Added command to delete the DB file when purge is done.

### CODE REVIEW

**Comment by venkatBala** [comment]
resolves #936

File: covalent_dispatcher/_cli/service.py [hunk 1]
```diff
@@ -386,20 +388,55 @@ def status() -> None:
...
+        click.echo("1. Stop the covalent server if running.")
+
+        for i, rem_path in enumerate(removal_list, start=2):
+            if os.path.isfile(rem_path):
+                click.echo(f"{i}. {rem_path} file will be DELETED")
+            else:
+                click.echo(f"{i}. {rem_path} directory will be DELETED")
+
+        if hard:
+            click.secho("Please make sure you are OK WITH THE DATABASE BEING DELETED.", fg="red")
```
Comment (wjcunningham7) [inline covalent_dispatcher/_cli/service.py:424]: Edit message to say "WARNING: All user data will be deleted."

File: covalent_dispatcher/_cli/service.py [hunk 2]
```diff
@@ -386,20 +388,55 @@ def status() -> None:
...
+
+        click.confirm("\nWould you still like to proceed?", abort=True)
```
Comment (wjcunningham7) [inline covalent_dispatcher/_cli/service.py:426]: Edit message to say "Would you like to proceed?"

File: covalent_dispatcher/_cli/service.py [hunk 3]
```diff
@@ -410,12 +447,11 @@ def logs() -> None:
...
-        while line:
-            click.echo(line.rstrip("\n"))
+        with open(UI_LOGFILE, "r") as f:
             line = f.readline()
-        f.close()
+            while line:
+                click.echo(line.rstrip("\n"))
+                line = f.readline()
     else:
         click.echo(f"{UI_LOGFILE} not found!. Server possibly purged!")
```
Comment (wjcunningham7) [inline covalent_dispatcher/_cli/service.py:456]: Edit message to say "{UI_LOGFILE} not found. Restart the server to create a new log file."

File: covalent_dispatcher/_cli/service.py [hunk 4]
```diff
@@ -386,20 +388,55 @@ def status() -> None:
...
+
+        click.secho(f"{''.join(['*'] * 21)} WARNING {''.join(['*'] * 21)}", fg="yellow")
+
+        click.echo("Purging will perform the following operations: ")
+
+        click.echo("1. Stop the covalent server if running.")
+
+        for i, rem_path in enumerate(removal_list, start=2):
+            if os.path.isfile(rem_path):
+                click.echo(f"{i}. {rem_path} file will be DELETED")
```
Comment (wjcunningham7) [inline covalent_dispatcher/_cli/service.py:419]: Remove capitalization

File: covalent_dispatcher/_cli/service.py [hunk 5]
```diff
@@ -386,20 +388,55 @@ def status() -> None:
 
 
 @click.command()
-def purge() -> None:
+@click.option("-H", "--hard", is_flag=True, help="Perform a hard purge, deleting the DB as well")
```
Comment (wjcunningham7) [inline covalent_dispatcher/_cli/service.py:391]: Help text: "Delete the dispatch database upon purge"

Also set `show_default=True` for these options.

File: covalent_dispatcher/_cli/service.py [hunk 6]
```diff
@@ -386,20 +388,55 @@ def status() -> None:
...
+@click.option("-y", "--yes", is_flag=True, help="Approve without showing the warning")
+def purge(hard: bool, yes: bool) -> None:
     """
-    Shutdown server and delete the cache and config settings.
+    Shuts down the server, deletes the cache, the logs, the config settings and if desired, the database.
```
Comment (wjcunningham7) [inline covalent_dispatcher/_cli/service.py:395]: Docstring text: "Purge Covalent from this system. This command is for developers."

File: covalent_dispatcher/_cli/service.py (hunk 5 above)
Comment (kessler-frost) [reply covalent_dispatcher/_cli/service.py]: Here they have mentioned that if the default is False then even `show_default` will not show the default value: https://click.palletsprojects.com/en/8.1.x/options/#basic-value-options

**Review by santoshkumarradha — APPROVED** [review]
Gtg from my side!

**Review by cjao — APPROVED** [review]

**Pull request merged**